### PR TITLE
Add simple benchmark for macros

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: tests/scripts/power-macro [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
```
Comment (liufengyun) [inline tests/scripts/power-macro:2]: A cleanup of `out` as an initial step?

File: tests/scripts/power-macro [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+mkdir "$PROG_HOME/tests/power-macro/out"
+
+dotc -d "$PROG_HOME/tests/power-macro/out" PowerMacro.scala
```
Comment (liufengyun) [inline tests/scripts/power-macro]: `dotc` is not available. You can change it to a command `compile`, which I'm going to provide.

File: tests/scripts/power-macro [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir "$PROG_HOME/tests/power-macro/out"
+
+dotc -d "$PROG_HOME/tests/power-macro/out" "$PROG_HOME/tests/PowerMacro.scala"
```
Comment (liufengyun) [inline tests/scripts/power-macro]: `dotc` is not available. You can change to `compile`, which I'm going to provide.

File: tests/scripts/power-macro [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+measure "$PROG_HOME/tests/PowerInlined-1.scala" -classpath "$PROG_HOME/tests/power-macro/out"
+measure "$PROG_HOME/tests/PowerInlined-1k.scala" -classpath "$PROG_HOME/tests/power-macro/out"
```
Comment (liufengyun) [inline tests/scripts/power-macro]: You also need to add a chart in [bin/charts.js](https://github.com/liufengyun/bench/blob/master/bin/charts.js)

File: tests/power-macro/PowerMacro.scala [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import scala.quoted.Expr
+
+object PowerMacro {
+
+  inline def power(inline n: Long, x: Double) = ~powerCode(n, '(x))
+
+  def powerCode(n: Long, x: Expr[Double]): Expr[Double] = {
+    println(s"powerCode($n)")
```
Comment (liufengyun) [inline tests/power-macro/PowerMacro.scala]: Better not print in benchmarks, the IO will cause big variance.

**Comment by nicolasstucki** [comment]
Done, now we just need the `compile` script

**Comment by liufengyun** [comment]
OK, I'm on it now and up to review soon.

File: tests/scripts/power-macro [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+mkdir "$PROG_HOME/tests/power-macro/out"
+
+compile -d "$PROG_HOME/tests/power-macro/out" "$PROG_HOME/tests/PowerMacro.scala"
```
Comment (liufengyun) [inline tests/scripts/power-macro]: The path for `PowerMacro.scala` is incorrect.

File: tests/scripts/power-macro [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+
+measure "$PROG_HOME/tests/PowerInlined-1.scala" -classpath "$PROG_HOME/tests/power-macro/out"
+measure "$PROG_HOME/tests/PowerInlined-1k.scala" -classpath "$PROG_HOME/tests/power-macro/out"
```
Comment (liufengyun) [inline tests/scripts/power-macro]: Same path problem for `PowerInlined-1k.scala` and `PowerInlined-1.scala`.

**Comment by liufengyun** [comment]
Now `compile` is implemented, please review @nicolasstucki , thanks.

**Comment by nicolasstucki** [comment]
Your commit LGTM

**Comment by liufengyun** [comment]
There are still two problems with the paths.

**Comment by nicolasstucki** [comment]
Done

**Comment by liufengyun** [comment]
Now there are some points, it's still running: http://dotty-bench.epfl.ch/

**Pull request merged**